### ISSUE TITLE
When updating schema on dataset there is no check for collisions with conformance rules

### ISSUE THREAD

**Issue by Zejnilovic** [AbsaOSS/enceladus#715]
## Describe the bug
If I have a schema with columns `a`,`b`,`c`, create a column `d` (string) using conformance rule and then update schema to contain `d` (timestamp) column of different kind and change it for the dataset, there is nothing stopping me and the mistake will be only evident in the middle of conformance when it tells me there already is column `d`.

## To Reproduce
Steps to reproduce the behavior OR commands run:
1. Have dataset `X` with columns `a`,`b`,`c`
2. Add conformance rule that creates columns `d`
3. Change schema of dataset `X` to have column `d` from raw. 
4. Run standardization and conformance
5. See error from conformance in console/logger/STDOUT

## Expected behaviour
When changing/updating schema check if it is possible depending on output columns of conformance rules.

**Comment by GeorgiChochov** [comment]
Yes, but we shouldn't disallow the schema from being changed. Let's add a warning on the dataset saying that it's in a `non-runnable` state.

**Comment by Zejnilovic** [comment]
Now we found out that this could lead to almost a cyclic issue where no rule can be deleted because rule dependency validation fails when trying to validate.
